theano/sandbox/gpuarray/conv_desc.c

```diff
@@ -0,0 +1,34 @@
+#section support_code_apply
+
+int conv_desc(PyArrayObject *filt_shp,
+              cudnnConvolutionDescriptor_t *desc) {
+  cudnnStatus_t err;
+  int pad[3] = {PAD_0, PAD_1, PAD_2};
+  int strides[3] = {SUB_0, SUB_1, SUB_2};
+  int upscale[3] = {1, 1, 1};
+
+#if BORDER_MODE == 0
+  pad[0] = *(npy_int64 *)PyArray_GETPTR1(filt_shp, 2) - 1;
+  pad[1] = *(npy_int64 *)PyArray_GETPTR1(filt_shp, 3) - 1;
+#if NB_DIMS > 2
+  pad[2] = *(npy_int64 *)PyArray_GETPTR1(filt_shp, 4) - 1;
+#endif
+#endif
+
+  if (PyArray_DIM(filt_shp, 0) - 2 != NB_DIMS) {
+    PyErr_Format(PyExc_ValueError, "Filter shape has too many dimensions: "
+                 "expected %d, got %lld.", NB_DIMS,
+                 (long long)PyArray_DIM(filt_shp, 0));
```

Comment:
FAIL only works in the init_code_struct, code and code_cleanup sections.  This is not one of those.

Also I do pass a function name to __init__ which places me under these conditions: http://deeplearning.net/software/theano/tutorial/extending_theano_c.html#main-function